crates/ark/src/lsp/completions/sources/common/subset.rs

```diff
@@ -0,0 +1,61 @@
+//
+// subset.rs
+//
+// Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+//
+//
+
+use tree_sitter::Node;
+use tree_sitter::Point;
+
+use crate::lsp::traits::point::PointExt;
+use crate::treesitter::NodeType;
+use crate::treesitter::NodeTypeExt;
+
+pub(crate) fn is_within_subset_delimiters(
+    x: &Point,
+    subset_node: &Node,
+    subset_type: &NodeType,
+) -> bool {
+    let (open, close) = match subset_type {
+        NodeType::Subset => ("[", "]"),
+        NodeType::Subset2 => ("[[", "]]"),
+        _ => std::unreachable!(),
+    };
+
+    let Some(arguments) = subset_node.child_by_field_name("arguments") else {
+        return false;
+    };
+
+    let n_children = arguments.child_count();
+
+    if n_children < 2 {
+        return false;
+    }
+
+    let Some(open_node) = arguments.child(1 - 1) else {
```

Comment:
Ah I didn't touch this function it just got moved. But I think it was to provide some symmetry between the first child / last child cases, i.e.

```
first_child = 1 - 1
last_child = n_children - 1
```

Rather than just `first_child = 0`

In retrospect it doesn't seem needed